src/gluonnlp/embedding/token_embedding.py

```diff
@@ -334,23 +337,23 @@ def _load_embedding_serialized(self, pretrained_file_path):
         deserialized_embedding = TokenEmbedding.deserialize(pretrained_file_path)
         if deserialized_embedding.unknown_token:
             # Some .npz files on S3 may contain an unknown token and its
-            # respective embedding. As a workaround, we assume that C.UNK_IDX
+            # respective embedding. As a workaround, we assume that UNK_IDX
             # is the same now as it was when the .npz was generated. Under this
             # assumption we can safely overwrite the respective token and
             # vector from the npz.
             if deserialized_embedding.unknown_token:
                 idx_to_token = deserialized_embedding.idx_to_token
                 idx_to_vec = deserialized_embedding.idx_to_vec
-                idx_to_token[C.UNK_IDX] = self.unknown_token
+                idx_to_token[UNK_IDX] = self.unknown_token
                 if self._init_unknown_vec:
                     vec_len = idx_to_vec.shape[1]
-                    idx_to_vec[C.UNK_IDX] = self._init_unknown_vec(shape=vec_len)
+                    idx_to_vec[UNK_IDX] = self._init_unknown_vec(shape=vec_len)
             else:
                 # If the TokenEmbedding shall not have an unknown token, we
                 # just delete the one in the npz.
-                assert C.UNK_IDX == 0
-                idx_to_token = deserialized_embedding.idx_to_token[C.UNK_IDX + 1:]
-                idx_to_vec = deserialized_embedding.idx_to_vec[C.UNK_IDX + 1:]
+                assert UNK_IDX == 0
```

Comment:
is this needed anymore?

Reply:
Currently the TokenEmbedding uses a different methods to initialize it's internals depending on from what the user wants to construct the TokenEmbedding. It is thus a bit ugly to remove the UNK_IDX dependency and may introduce bugs easily.

The best way forward is to refactor the different place in TokenEmbedding that currently must modify and construct the internal attributes, making them static and extending the constructor to support initializing a TokenEmbedding given the list of tokens and the ndarray of embeddings.

I have prepared some code, but let's discuss it in a separate PR.

Reply:
Thanks for the explanation. I was referring to only the assert